python/cudf/cudf/_lib/json.pyx

```diff
@@ -146,6 +167,71 @@ cpdef read_json(object filepaths_or_buffers,
 
     return df
 
+cpdef chunked_read_json(object filepaths_or_buffers,
+                        object dtype,
+                        object compression,
+                        bool keep_quotes,
+                        bool mixed_types_as_string,
+                        bool prune_columns,
+                        object on_bad_lines,
+                        int chunk_size=100_000_000):
+    """
+    Cython function to call into libcudf API, see `read_json`.
+
+    See Also
+    --------
+    cudf.io.json.read_json
+    cudf.io.json.to_json
+    """
+    cdef size_type c_range_size = (
+        chunk_size if chunk_size is not None else 0
+    )
+    cdef json_reader_options opts = _setup_json_reader_options(
+        filepaths_or_buffers, dtype, compression, keep_quotes,
+        mixed_types_as_string, prune_columns, on_bad_lines,
+        True, 0, c_range_size
+    )
+
+    # Read JSON
+    cdef cudf_io_types.table_with_metadata c_result
+    final_columns = []
+    meta_names = None
+    i = 0
+    while True:
+        opts.set_byte_range_offset(c_range_size * i)
+        opts.set_byte_range_size(c_range_size)
+
+        try:
+            with nogil:
+                c_result = move(libcudf_read_json(opts))
+        except (ValueError, OverflowError):
```

Comment:
Just curious if we could use some logic such as `_has_next()` in PQ chunked reader to break this loop instead of this exception?

Reply:
I don't like that we're catching the exception from a datasource here. The memory mapping is very much an implementation detail.
How expensive would it be to get the total source(s) size? Then we can loop until all of it is read.

Reply:
It is expensive, we basically have to seek to the end of file to get the size of all data sources. For remote data-sources it get's complicated to properly perform seek too.

Reply:
> Just curious if we could use some logic such as `_has_next()` in PQ chunked reader to break this loop instead of this exception?

We basically call into libcudf layer for that, is there any such provision for json reader in libcudf?

Reply:
FWIW, we already need the file size(s) to read JSON input(s). With the current implementation of the low memory JSON reader, we get the size of each input file for each byte range, so getting the sizes one more time to have a cleaner loop would not  add much.